docs: add an overview of contribution guidelines (#43)
autowarefoundation/autoware-documentation: docs/contributing/index.md

# Contributing

This page explains how to contribute to Autoware.

First, confirm the [Code of conduct](#contribution-workflows) section for the general manners in contributions.

See the [Contribution workflows](#contribution-workflows) section for the workflows.  
Also, confirm the following our guidelines depending on your situation:

- [Coding guidelines](coding-guidelines/index.md): when you write source code
- [Testing guidelines](testing-guidelines/index.md): when you write tests
- [Documentation guidelines](documentation-guidelines/index.md): when you write documents
- [Pull request guidelines](pull-request-guidelines/index.md): when you submit pull requests
- [Discussion guidelines](discussion-guidelines/index.md): when you discuss with other contributors
- [Support guidelines](../help/support-guidelines.md): when you need any help

It is recommended to read the [Contributing pages in ROS 2 Docs](https://docs.ros.org/en/rolling/Contributing.html) as well because Autoware's guidelines generally depend on them.

For general information about OSS contributions, [Open Source Guides](https://opensource.guide/) by GitHub is a good resource.

> ## Code of conduct

Author: This sections was moved from https://github.com/autowarefoundation/autoware/pull/29.

# Contributing

This page explains how to contribute to Autoware.

> First, confirm the [Code of conduct](#contribution-workflows) section for the general manners in contributions.

Reviewer: ```suggestion
First, conform to the [Code of conduct](#contribution-workflows) section for the general manners in contributions.
```

See the [Contribution workflows](#contribution-workflows) section for the workflows.  

> Also, confirm the following our guidelines depending on your situation:

Reviewer: ```suggestion
Also, make sure the contribution is in line with the following guidelines:
```

> - [Coding guidelines](coding-guidelines/index.md): when you write source code
> - [Testing guidelines](testing-guidelines/index.md): when you write tests
> - [Documentation guidelines](documentation-guidelines/index.md): when you write documents
> - [Pull request guidelines](pull-request-guidelines/index.md): when you submit pull requests
> - [Discussion guidelines](discussion-guidelines/index.md): when you discuss with other contributors
> - [Support guidelines](../help/support-guidelines.md): when you need any help

Reviewer: ```suggestion
- [Coding guidelines](coding-guidelines/index.md): for contributing source code
- [Testing guidelines](testing-guidelines/index.md): for contributing tests
- [Documentation guidelines](documentation-guidelines/index.md): for contributing documents
- [Pull request guidelines](pull-request-guidelines/index.md): for submitting pull requests
- [Discussion guidelines](discussion-guidelines/index.md): for discussing with other contributors
- [Support guidelines](../help/support-guidelines.md): for finding the right channel when you need help
```

It is recommended to read the [Contributing pages in ROS 2 Docs](https://docs.ros.org/en/rolling/Contributing.html) as well because Autoware's guidelines generally depend on them.

For general information about OSS contributions, [Open Source Guides](https://opensource.guide/) by GitHub is a good resource.

!!!info

>     If you want to apply a new or different rule, you can open a new discussion thread and propose it.

Reviewer: ```suggestion
    If you want to make change to these or any other rules/guidelines, you can open a new discussion thread and propose it.
```

Author: `make change` -> `make a change` or `make changes`?

Author: How about this?

```
If you want to make a change to any guidelines in this documentation site, you can open a new discussion thread and propose it.
```

Reviewer: `make changes` is better.

`to any guidelines` here the `any` is redundant.

I've updated it, does it work for you?

Author: Yeah! Thank you.

## Code of conduct

We welcome contributions from everyone.  
To ensure our community stays open and healthy, we adhere to the [Contributor Covenant](https://www.contributor-covenant.org/), a widely used [code of conduct](https://github.com/autowarefoundation/autoware/blob/main/CODE_OF_CONDUCT.md) adopted by many other communities listed [here](https://www.contributor-covenant.org/adopters/).

Everyone participating in the Autoware community is expected to follow the code of conduct.  
If someone in the community happens to be violating these terms, let the maintainers know, and we will address it as soon as possible.

!!!warning

    Note that contributions that ignore guidelines are not contributions but nuisances.

## Contribution workflows

There are several types of contributions and the workflows differ between the types.
See the description for your contribution type.

### Discussions

You can contribute to Autoware by facilitating discussions:

- Find a bug and open a new issue.
- Propose an idea of features that will enhance Autoware.
- Join an existing discussion and express your opinion.

> - Organize discussion so far for others contributors.

Reviewer: ```suggestion
- Organize discussions to get opinions of other contributors.
```
I think you've meant this?

Author: @xmfcx What I wanted to say is:
- Sometimes people say only what you like and make others confused.
- In that case, we need some facilitators.

Reviewer: I've added ` about their ideas.` does this work?

Author: Hmm, I personally prefer a simpler one.
Even `Organize discussions for other contributors.` looks good to me.

FYI: https://developers.google.com/style/translation?hl=en#write-short,-clear,-and-precise-sentences

Reviewer: Sure, I've updated it.

> - Answer an question and support other contributors.

Reviewer: ```suggestion
- Answer questions and support other contributors.
```

See our [support guidelines](../help/support-guidelines.md) for the detailed steps.

### Pull requests

> You can directly submit a pull request if it is small like the following:

Reviewer: ```suggestion
You can submit a pull request directly (without opening an issue) if it is small like the following:
```

Author: @xmfcx Just out of curiosity, what's the difference between `You can directly submit a pull request` and `You can submit a pull request directly`?

Most Japanese cannot understand the nuance. :cry:

Reviewer: Not much difference. Your usage is right. But if we add `(without opening an issue)` then we must move it to the end or it will look weird.

Wrong:
- You can `without opening an issue` submit a pull request
Right:
- You can submit a pull request `without opening an issue`

Maybe you could add commas and would be correct but putting it to end is better:
- You can, `without opening an issue`, submit a pull request

I understand your pain because turkish is very similar to japanese way of constructing sentences, we put the predicates to the end. And adverbs come before the predicates. But in english they put adverbs after the predicates most of the time.

Extra info:
> Noun In “arrived early,” “runs slowly,” “stayed home,” and “works hard” the words “early,” “slowly,” “home,” and “hard” are adverbs. https://www.merriam-webster.com/dictionary/adverb

Author: I understand! Thank you for the detailed explanation!

Reviewer: > Kare wa toshokan ni todomatta.

Literal translation would be:

> He library <- in stayed.

But right order is:

> He stayed in library.

> - Update a small part of documents.
> - Fix spelling mistakes.
> - Fix CI failures.
> - Fix warnings detected by compilers or analysis tools.

Reviewer: ```suggestion
- Update a small part of documents.
- Fix spelling mistakes.
- Fix CI failures.
- Fix warnings detected by compilers or analysis tools.
- Make small changes to a single package.
```
